city_scrapers/spiders/cook_justice_advisory.py

```diff
@@ -0,0 +1,143 @@
+            for href in response.xpath('//a[contains(text(), "JAC Council")]')
+            .css("a::attr(href)")
+            .extract()
+        ]
+
+    def _parse_location(self, response):
+        """
+        Parse or generate location. Url, latitude and longitude are all
+        optional and may e more trouble than they're worth to collect.
+        """
+        address = response.xpath(
+            '//div[@class="field event-location"]/descendant::*/text()'
+        ).extract()
+        for word in ["Location:", ", ", " "]:
+            address.remove(word)
+        address = " ".join(address)
+        return {
+            "address": address,
+            "name": "",
+        }
+
+    def _parse_all_day(self, response):
+        """
+        Parse or generate all-day status. Defaults to false.
+        """
+        date = response.xpath(
+            '//span[@class="date-display-single"]/descendant-or-self::*/text()'
+        ).extract()
+        date = "".join(date).upper()
+        return "ALL DAY" in date
+
+    def _parse_title(self, response):
+        """Parse or generate event"""
+        title = response.css("h1::text").extract()
+        if "Special" in title:
+            return "Special JAC Council Meeting"
+        elif "JAC Council Meeting" in title:
+            return "JAC Council Meeting"
+        else:
+            return title
+
+    def _parse_description(self, response):
+        """Parse or generate event description."""
+        category_field = response.xpath(
+            "//div[contains(., 'Category:') and contains(@class, 'field-label')]"
+        )
+        field_items = category_field.xpath(
+            "./following::div[contains(@class, 'field-items')]"
+        )
+        return " ".join(
+            field_items.xpath(".//p/text()").extract()
+            + field_items.xpath(".//strong/text()").extract()
+        ).strip()
+
+    def _parse_start(self, response):
+        """Parse start date and time"""
+        start = response.xpath(
+            '//span[@class="date-display-single"]/descendant-or-self::*/text()'
+        ).extract()
+        start = "".join(start).upper()
+        start = start.split(" TO ")[0].strip()
+        start = start.replace("(ALL DAY)", "12:00AM")
+
+        return datetime.strptime(start, "%B %d, %Y %I:%M%p")
+
+    def _parse_end(self, response):
+        """Parse end date and time"""
+        date = response.xpath(
+            '//span[@class="date-display-single"]/descendant-or-self::*/text()'
+        ).extract()
+        date = "".join(date).upper()
+        date.replace("(ALL DAY)", "TO 11:59PM")
+        start_end = date.split(" TO ")
+
+        if len(start_end) < 2:
+            return
+
+        end_time = start_end[1]
+        date = start_end[0][: start_end[0].rindex(" ")]
+        return datetime.strptime("{} {}".format(date, end_time), "%B %d, %Y %I:%M%p")
+
+    def _parse_links(self, response):
+        files = response.css("span.file a")
```

Comment:
This is only going to get documents on the event page itself, but we'll want the minutes and agendas if they match up to an event that are on https://www.cookcountyil.gov/service/justice-advisory-council-meetings

We'll want to create a mapping of months and years to links for those meetings and then reference that when pulling from events pages so that the agendas and minutes can be associated. Here's one example of something similar https://github.com/City-Bureau/city-scrapers/blob/19fd52d76aa314734f620c4dafa9be0a0228ba81/city_scrapers/spiders/il_labor.py#L28

Reply:
Should cancelled meetings also get scraped? There is an agenda link associated with the cancelled meeting in March.

Reply:
Unrelated to the last comment, but is there any special way virtual meeting addresses should be handled? Right now the spider returns it as 'Microsoft Teams Chicago IL 60602'.

Reply:
Yep, we'll want to know when meetings are cancelled. And if it's a virtual location I think that probably goes better in the description and can be left blank in the location

Reply:
I believe we have implemented all of the comments that you left, could you take another look at our code? Thanks!